Cleanup staticcheck issues for apiextension

RainbowMango opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Cleanup issues identified by staticcheck in #81189

**Which issue(s) this PR fixes**:
ref #81657

**Special notes for your reviewer**:

In order to review more convenient, split with check types.
For more details about check description, please refer to http://staticcheck.io/docs/checks.

Commit: Dealing with unused functions/variables/types. (staticcheck U1000)
```
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/webhook_converter.go:116:28: func (*webhookConverter).hasConversionReviewVersion is unused (U1000)
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/goopenapi_test.go:34:5: var nullTypeRE is unused (U1000)
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/validation_test.go:228:6: func invalidIndex is unused (U1000)
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/validation_test.go:231:6: func unsupported is unused (U1000)
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/validation_test.go:234:6: func immutable is unused (U1000)
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/validation_test.go:237:6: func forbidden is unused (U1000)
vendor/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:945:2: field options is unused (U1000)
```

Commit: Dealing with value never used issue. (staticcheck SA4006)
```
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler_test.go:98:10: this value of err is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go:133:4: this value of crd is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go:345:18: this value of found is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/basic_test.go:527:2: this value of noxuDefinition is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/basic_test.go:830:24: this value of err is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/basic_test.go:836:24: this value of err is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:274:4: this value of obj is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/fixtures/server.go:36:18: this value of err is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:539:4: this value of createdNoxuInstance is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:585:2: this value of crd is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:616:3: this value of crd is never used (SA4006)
vendor/k8s.io/apiextensions-apiserver/test/integration/validation_test.go:650:3: this value of crd is never used (SA4006)
```

Commit: Dealing with concurrency issue. (staticcheck SA2002 SA4010)
```
vendor/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/pruning/algorithm_test.go:639:21: this result of append is never used, except maybe in other appends (SA4010)
vendor/k8s.io/apiextensions-apiserver/test/integration/change_test.go:67:2: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
vendor/k8s.io/apiextensions-apiserver/test/integration/change_test.go:99:3: the goroutine calls T.Fatal, which must be called in the same goroutine as the test (SA2002)
vendor/k8s.io/apiextensions-apiserver/test/integration/change_test.go:119:3: the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
```

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

RainbowMango committed:
Fix staticcheck issues:
Dealing with unused functions/variables/types. (staticcheck U1000)
Dealing with value never used issue. (staticcheck SA4006)
Dealing with concurrency issue. (staticcheck SA2002 SA4010)
Remove packages from staticcheck failure files: apiextensions-apiserver

RainbowMango commented:
/assign @sttts

hzxuzhonghu commented:
/ok-to-test

hzxuzhonghu commented:
/lgtm

sttts commented on staging/src/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go:
```diff
@@ -130,7 +130,7 @@ func (c *CRDFinalizer) sync(key string) error {
 		apiextensions.SetCRDCondition(crd, cond)
 		if deleteErr != nil {
-			crd, err = c.crdClient.CustomResourceDefinitions().UpdateStatus(crd)
+			_, err = c.crdClient.CustomResourceDefinitions().UpdateStatus(crd)
```
make it `if _, err := ...; err != nil {`

sttts commented on staging/src/k8s.io/apiextensions-apiserver/test/integration/change_test.go:
```diff
@@ -75,7 +75,8 @@ func TestChangeCRD(t *testing.T) {
 			if err != nil {
-				t.Fatal(err)
+				t.Error(err)
+				return
```
continue. And move the Sleep to the beginning of the loop. Same below.

sttts commented on staging/src/k8s.io/apiextensions-apiserver/test/integration/conversion/conversion_test.go:
```diff
@@ -271,7 +270,7 @@ func validateStorageVersion(t *testing.T, ctc *conversionTestContext) {
 			ctc.setAndWaitStorageVersion(t, "v1beta2")
 
-			obj, err = client.Get(obj.GetName(), metav1.GetOptions{})
+			_, err = client.Get(obj.GetName(), metav1.GetOptions{})
```
`if _, err := ...; err != nil {`

sttts commented on staging/src/k8s.io/apiextensions-apiserver/test/integration/fixtures/server.go:
```diff
@@ -53,7 +56,7 @@ users:
     username: test
 `)
-	fakeKubeConfig.Close()
+	_ = fakeKubeConfig.Close()
```
how ugly.

RainbowMango commented:
Fixed comments and squash commits.
@sttts

RainbowMango commented:
/retest

RainbowMango commented on staging/src/k8s.io/apiextensions-apiserver/test/integration/subresources_test.go:
```diff
@@ -536,8 +536,7 @@ func TestValidateOnlyStatus(t *testing.T) {
 			}
-			createdNoxuInstance, err = noxuResourceClient.UpdateStatus(createdNoxuInstance, metav1.UpdateOptions{})
-			if err == nil {
+			if _, err = noxuResourceClient.UpdateStatus(createdNoxuInstance, metav1.UpdateOptions{}); err != nil {
```
I made a stupid mistake here.
Stop using `if err = xxx; err != nil` ...since here we expect nil not error.

RainbowMango replied:
Fixed.

RainbowMango commented:
/retest

sttts commented on staging/src/k8s.io/apiextensions-apiserver/test/integration/change_test.go:
```diff
@@ -73,9 +73,12 @@ func TestChangeCRD(t *testing.T) {
 			noxuDefinitionToUpdate, err := apiExtensionsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(noxuDefinition.Name, metav1.GetOptions{})
 			if err != nil {
-				t.Fatal(err)
+				t.Error(err)
```
remind me why `t.Fatal` was bad here?

RainbowMango replied:
"the goroutine calls T.Fatal, which must be called in the same goroutine as the test"

RainbowMango replied:
It should be this: "the goroutine calls T.Fatal, which must be called in the same goroutine as the test"

RainbowMango replied:
Double checked.  You can try this command: 
`hack/verify-staticcheck.sh vendor/k8s.io/apiextensions-apiserver/test/integration`

sttts commented:
/lgtm
/approve

RainbowMango committed:
Fix a new staticcheck issue.
vendor/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go:167:2: this value of crd is never used (SA4006)

RainbowMango commented:
Rebase master and fix a new issue.
```
vendor/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go:167:2: this value of crd is never used (SA4006)
```

RainbowMango commented:
/test pull-kubernetes-node-e2e

RainbowMango commented:
/test pull-kubernetes-node-e2e

```
W0925 03:46:25.878] subprocess.CalledProcessError: Command '('kubetest', '--dump=/workspace/_artifacts', '--gcp-service-account=/etc/service-account/service-account.json', '--up', '--down', '--test', '--deployment=node', '--provider=gce', '--cluster=bootstrap-e2e', '--gcp-network=bootstrap-e2e', '--gcp-project=k8s-jkns-pr-node-e2e', '--gcp-zone=us-west1-b', '--node-test-args=--kubelet-flags="--cgroups-per-qos=true --cgroup-root=/"', '--node-tests=true', '--test_args=--nodes=8 --focus="\\[NodeConformance\\]" --skip="\\[Flaky\\]|\\[Slow\\]|\\[Serial\\]" --flakeAttempts=2', '--timeout=65m', '--node-args=--image-config-file=/workspace/test-infra/jobs/e2e_node/image-config.yaml')' returned non-zero exit status 1
```

RainbowMango commented:
Got the logs but don't know why. It may be related to signal handler.
```
I0925 04:16:35.562] E0925 04:16:27.264367    1295 services.go:88] Failed to stop services: error stopping "services": waitid: no child processes
```

/test pull-kubernetes-node-e2e

RainbowMango commented:
It should be fine now.
/test pull-kubernetes-node-e2e

hzxuzhonghu commented:
/lgtm
